src/EFCore.PG/Query/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -344,11 +344,17 @@ protected override Expression VisitSqlUnary(SqlUnaryExpression sqlUnaryExpressio
                 {
                 case SqlConstantExpression:
                 case SqlParameterExpression:
+                case SqlUnaryExpression { OperatorType: ExpressionType.Convert }:
+                case ColumnExpression:
                 case SqlFunctionExpression:
                 case ScalarSubqueryExpression:
-                    var storeType = sqlUnaryExpression.TypeMapping.StoreType;
-                    if (storeType == "integer")
-                        storeType = "INT";  // Shorthand that looks better in SQL
+                    var storeType = sqlUnaryExpression.TypeMapping.StoreType switch
+                    {
+                        "integer" => "INT",
```

Comment:
Only because it's been this way... There's already lots of inconsistency around upper/lower case in type and function names... We can look at this separately (don't want to cause even more test changes in this PR :/)